hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/codec/OMDBDefinitionBase.java

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.codec;
+
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.COMPACTION_LOG_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.DELEGATION_TOKEN_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.META_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.MULTIPART_INFO_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.PREFIX_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.PRINCIPAL_TO_ACCESS_IDS_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.S3_SECRET_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.SNAPSHOT_INFO_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.SNAPSHOT_RENAMED_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.TENANT_ACCESS_ID_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.TENANT_STATE_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.TRANSACTION_INFO_TABLE_DEF;
+import static org.apache.hadoop.ozone.om.codec.OMDBDefinition.USER_TABLE_DEF;
+
+import org.apache.hadoop.hdds.utils.TransactionInfo;
+import org.apache.hadoop.hdds.utils.db.DBColumnFamilyDefinition;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+import org.apache.hadoop.ozone.om.helpers.OmDBAccessIdInfo;
+import org.apache.hadoop.ozone.om.helpers.OmDBTenantState;
+import org.apache.hadoop.ozone.om.helpers.OmDBUserPrincipalInfo;
+import org.apache.hadoop.ozone.om.helpers.OmDirectoryInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmMultipartKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmPrefixInfo;
+import org.apache.hadoop.ozone.om.helpers.OmVolumeArgs;
+import org.apache.hadoop.ozone.om.helpers.RepeatedOmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.S3SecretValue;
+import org.apache.hadoop.ozone.om.helpers.SnapshotInfo;
+import org.apache.hadoop.ozone.security.OzoneTokenIdentifier;
+import org.apache.hadoop.ozone.storage.proto.OzoneManagerStorageProtos;
+import org.apache.ozone.compaction.log.CompactionLogEntry;
+
+/**
+ * Interface for abstracting access to OM RocksDB table definitions.
+ * <p>
+ * This interface provides methods to retrieve {@link DBColumnFamilyDefinition} instances
+ * for all tables used by the Ozone Manager (OM) or Recon, enabling shared
+ * initialization logic across different OM DB implementations.
+ * </p>
+ *
+ * <p>Example usage:</p>
+ * <pre>{@code
+ *   OmDBDefinitionBase defs = OMDBDefinition.INSTANCE;
+ *   DBColumnFamilyDefinition<?, ?> userTableDesc = defs.getUserTableDef();
+ * }</pre>
+ *
+ * Implementations:
+ * <ul>
+ *   <li>{@link OMDBDefinition}</li>
+ *   <li>{@link ReconOmDBDefinition}</li>
```

Comment:
```
[WARNING] Javadoc Warnings
[WARNING] hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/codec/OMDBDefinitionBase.java:69: warning - Tag @link: reference not found: ReconOmDBDefinition
```

Cannot link to `ReconOmDBDefinition` (as Recon depends on OM, not the other way around).  Please use `@code` instead.